app/src/main/java/my/noveldokusha/ui/screens/main/MainActivity.kt

```diff
@@ -85,6 +88,30 @@ open class MainActivity : BaseActivity() {
                 }
             }
         }
+
+        val action: String = intent.action!!
+        val type: String? = intent.type
+
+        if (Intent.ACTION_SEND == action && type != null) {
+            if ("application/epub+zip" == type) {
+                handleSharedEpub(intent)
+            }
+        } else if (Intent.ACTION_VIEW == action) {
+            handleViewedEpub(intent)
+        }
```

Comment:
extract to method and use `when` switch
```kotlin
private fun handleIntent(intent){
   val action = intent.action ?: return
   val type = intent.type
   
   when (action) {
     Intent.ACTION_SEND -> {
        if (type == "application/epub-zip") {
            handle....
         }
     }
    
     Intent.ACTION_VIEW -> {
       .....
    }

   }
  
}

Reply:
Ok, I extracted a handleIntent method now. Looks ok?